tests/test_updater_with_simulator.py

```diff
@@ -257,6 +259,50 @@ def test_not_loading_targets_twice(self, wrapped_open: MagicMock) -> None:
         updater.get_targetinfo("somepath")
         wrapped_open.assert_not_called()
 
+    @patch.object(builtins, "open", wraps=builtins.open)
+    def test_expired_metadata(self, wrapped_open: MagicMock) -> None:
+        updater = self._run_refresh()
+
+        past_datetime = datetime.utcnow().replace(microsecond=0) - timedelta(days=5)
+
+        self.sim.timestamp.expires = past_datetime
+        self.sim.snapshot.expires = past_datetime
+        self.sim.targets.expires = past_datetime
+
+        # Add targets to repository
+        self.sim.targets.version += 1
+        self.sim.add_target("targets", b'some content', self.targets_dir)
+        self.sim.update_snapshot()
+
+        # Clean up calls to open during refresh()
+        wrapped_open.reset_mock()
+
+        # Make a successful update of valid metadata which stores it in cache
+        updater.get_targetinfo(self.targets_dir)
+
+        # Create a new updater and perform a second update while
+        # the metadata is already stored in cache (metadata dir)
+        updater = Updater(
+            self.metadata_dir,
+            "https://example.com/metadata/",
+            self.targets_dir,
+            "https://example.com/targets/",
+            self.sim,
+        )
+
+        with self.assertRaises(ExpiredMetadataError):
+            updater.get_targetinfo(self.targets_dir)
+
+        # Test that an expired timestamp/snapshot/targets
+        # when loaded from cache is not stored as final
+        # but is used to verify the new timestamp
+        wrapped_open.assert_has_calls([
+            call(os.path.join(self.metadata_dir, "root.json"), "rb"),
+            call(os.path.join(self.metadata_dir, "timestamp.json"), "rb"),
+        ])
```

Comment:
When reading the changes made in this pr I am comparing them to @sechkova comment here: https://github.com/theupdateframework/python-tuf/issues/1681#issuecomment-977851021.
It's not clear to me after you finish step 2 which steps do you cover with those asserts.
@ivanayov can you give more details on that?

Reply:
@MVrachev This PR is for step 4 only. The others are covered in #1703 .

Reply:
Step 5 is already covered in tests, so we discussed with @jku that it's not necessary to repeat it here.

Reply:
I will look at it today a little later.

Reply:
Thanks, I've edited the description.